src/zql/query/entity-query.ts

```diff
@@ -12,10 +16,27 @@ import {EntitySchema} from '../schema/entity-schema.js';
 import {AggArray, Aggregate, Count, isAggregate} from './agg.js';
 import {Statement} from './statement.js';
 
-type FieldValue<
+type NotUndefined<T> = Exclude<T, undefined>;
+
+export type FieldValue<
   S extends EntitySchema,
   K extends Selectable<S>,
-> = S['fields'][K] extends Primitive | undefined ? S['fields'][K] : never;
+  Op extends SimpleOperator,
+> = S['fields'][K] extends Primitive | undefined
+  ? Op extends InOps
+    ? NotUndefined<S['fields'][K]>[]
```

Comment:
I decided to not allow undefined in these cases.

I guess it could be argued that we should allow it for `=` and `!=`?

Reply:
I think we should think about what we want here when it comes to representing optional values in PSQPL.

Reply:
`null` is another thing we need to think about. Unfortunately SQL has odd rules around `null` when it comes to equality.

```
sqlite> create table foo (a);
sqlite> insert into foo values (null);
sqlite> .mode qbox
sqlite> select * from foo;
┌──────┐
│  a   │
├──────┤
│ NULL │
└──────┘
sqlite> select * from foo where a = null;
sqlite> select * from foo where a != null;
sqlite> select * from foo where a IS null;
┌──────┐
│  a   │
├──────┤
│ NULL │
└──────┘
sqlite> select * from foo where a IS NOT null;
sqlite>
```